eva-tools/src/test/java/embl/ebi/variation/eva/vcfDump/regionUtils/IntersectingRegionsMergerTest.java

```diff
@@ -0,0 +1,67 @@
+package embl.ebi.variation.eva.vcfdump.regionutils;
+
+import org.junit.Test;
+import org.opencb.biodata.models.feature.Region;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by pagarcia on 09/05/2016.
+ */
+public class IntersectingRegionsMergerTest {
+
+    @Test
+    public void testNoIntersectingRegionList() {
+        List<Region> nonIntersectingRegions = Arrays.asList(new Region("1", 100, 200), new Region("1", 300, 500), new Region("2", 150, 250));
+        List<Region> mergedRegions = nonIntersectingRegions.stream().collect(new IntersectingRegionsMerger());
+        assertEquals(nonIntersectingRegions, mergedRegions);
+    }
+
+    @Test
+    public void testSortedOutput() {
+        List<Region> unsortedRegions = Arrays.asList(new Region("1", 300, 500), new Region("2", 150, 250), new Region("1", 100, 200));
+        List<Region> mergedRegions = unsortedRegions.stream().collect(new IntersectingRegionsMerger());
+        assertEquals(mergedRegions, Arrays.asList(new Region("1", 100, 200), new Region("1", 300, 500), new Region("2", 150, 250)));
```

Comment:
The expected value should go first, otherwise JUnit could report errors in a bit confusing way :)